specify/specify7
# Don't trigger unload protect when setting LatLon plugin values

Fixes https://github.com/specify/specify7/issues/3259#issuecomment-1699725102

The issue occurred when setting the value for `lat1text`, `lat2text`, `long1text`, and `long2text` for the LatLonUI Plugin. 

Which occured at this line:
https://github.com/specify/specify7/blob/312139ad3ee9290dd00500e764dd0f0c36f8b026/specifyweb/frontend/js_src/lib/components/FormPlugins/LatLongUi.tsx#L109

To highlight the issue, I will use an example from the `sdnhmpaleo_3_31_23` database. 
(https://sdnhmpaleo33123-v79-dev.test.specifysystems.org/specify/view/locality/24618/)

Specify tries to convert the value of  in the database into a valid latitude/longitude value. 

Here are the values in the database. 
```sql
+-----------------+-------------------+
| lat1text        | long1text         |
+-----------------+-------------------+
| 32.810579Â° N   | 117.2229226Â° W   |
+-----------------+-------------------+
```
Specify trims lat1text and long1text to `32.810579 N` and `117.2229226 W` (respectively), and sets the value of the fields to the trimmed version, which triggers unload protect. 
I have made the setting of these values "silent" which means they will not trigger unload protect. 

TO TEST:
Make sure the resource's unload protect is not triggered when being initialized, but ensure that manually editing every component still properly triggers unload protect.
